keras/applications/regnet.py

```diff
@@ -0,0 +1,1349 @@
+           filters_out,
+           group_width,
+           stride=1,
+           name=""):
+  """Implementation of X Block. 
+  Reference: [Designing Network Design Spaces](https://arxiv.org/abs/2003.13678)
+
+  Args:
+    inputs: input tensor
+    filters_in: filters in the input tensor
+    filters_out: filters in the output tensor
+    group_width: group width
+    stride: stride
+    name: name prefix
+
+  Return:
+    Output tensor of the block 
+  """
+  def apply(inputs):
+    if filters_in != filters_out and stride == 1:
+      raise ValueError("""Input filters and output filters are not equal for stride
+                          1. Please check inputs and try again.""")
+
+    # Declare layers
+    groups = filters_out // group_width
+    
+    if stride != 1:
+      skip = layers.Conv2D(filters_out, (1, 1),
+                          strides=stride, use_bias=False,
+                          name=name + "_skip_1x1")(inputs)
+      skip = layers.BatchNormalization(
+          momentum=0.9, epsilon=1e-5, name=name + "_skip_bn")(skip)
+      conv_3x3 = layers.Conv2D(filters_out, (3, 3), use_bias=False, strides=stride,
+                              groups=groups, name=name + "_conv_3x3")
+    else:
+      skip = inputs
+      conv_3x3 = layers.Conv2D(filters_out, (3, 3), use_bias=False,
+                              groups=groups, name=name + "_conv_3x3")
+    
+      # Build block
+    # conv_1x1_1
+    x = layers.Conv2D(filters_out, (1, 1), use_bias=False,
+                      name=name + "_conv_1x1_1")(inputs)
+    x = layers.BatchNormalization(
+        momentum=0.9, epsilon=1e-5, name=name + "_conv_1x1_1_bn")(x)
+    x = layers.ReLU(name=name + "_conv_1x1_1_relu")(x)
+
+    # # conv_3x3
+    x = conv_3x3(x)
+    x = layers.BatchNormalization(
+        momentum=0.9, epsilon=1e-5, name=name + "_conv_3x3_bn")(x)
+    x = layers.ReLU(name=name + "_conv_3x3_relu")(x)
+
+    # conv_1x1_2
+    x = layers.Conv2D(filters_out, (1, 1), use_bias=False,
+                      name=name + "_conv_1x1_2")(x)
+    x = layers.BatchNormalization(
+        momentum=0.9, epsilon=1e-5, name=name + "_conv_1x1_2-bn")(x)
+
+    x = layers.ReLU(name=name + "_exit_relu")(x + skip)
+
+    return x
+  return apply
+
+
+def YBlock(filters_in,
+           filters_out,
+           group_width,
+           stride=1,
+           se_ratio=0.25,
+           name=""):
+  """Implementation of Y Block. 
+  Reference: [Designing Network Design Spaces](https://arxiv.org/abs/2003.13678)
+
+  Args:
+    inputs: input tensor
+    filters_in: filters in the input tensor
+    filters_out: filters in the output tensor
+    group_width: group width
+    stride: stride
+    se_ratio: expansion ration for Squeeze and Excite block
+    name: name prefix
+
+  Return:
+    Output tensor of the block 
+  """
+  def apply(inputs):
+    if filters_in != filters_out and stride == 1:
+      raise ValueError("""Input filters and output filters are not equal for stride
```

Comment:
In error message, specify:

- What the user passed
- What was expected instead
- How to fix it


Here, specify filters_in, filters_out, and stride, and suggest better argument values